compact.go

```diff
@@ -652,6 +692,12 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 	}
 
 	for set.Next() {
+		select {
+		case <-c.ctx.Done():
+			return nil
```

Comment:
ditto? The common practice is to return `ctx.Err` sometimes as well. It is easy then to filter out later on if you want

Reply:
In our case `ErrCompactionCanceled` would do I guess.

Reply:
I refactored to use `ctx.Err` , much better, thanks!